drivers/st25r3911b/st25r3911b_spi.c

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+
+#include <spi.h>
+#include <device.h>
+#include <soc.h>
+#include <gpio.h>
+#include <logging/log.h>
+
+#include "st25r3911b_spi.h"
+#include "st25r3911b_reg.h"
+
+LOG_MODULE_REGISTER(st25r3911b, CONFIG_ST25R3911B_LIB_LOG_LEVEL);
+
+static struct device *spi_dev;
+static struct device *gpio_dev;
+
+#define ST25R3911B_READ_REG(_reg) (0x40 | (_reg))
+#define ST25R3911B_WRITE_REG(_reg) (~0xC0 & (_reg))
+#define ST25R3911B_DIRECT_CMD(_cmd) (0xC0 | (_cmd))
+#define ST25R3911B_FIFO_READ 0xBF
+#define ST25R3911B_FIFO_WRITE 0x80
+
+#define REG_CNT 0x3F
+
+#define _DO_GPIO_PORT_NUM(_num) \
+	 DT_GPIO_P ## _num ## _DEV_NAME
+
+#define GPIO_PORT_NUM(_num) \
+	_DO_GPIO_PORT_NUM(_num)
+
+/* Timing defined by spec. */
+#define T_NCS_SCLK 1
+
+/* SPI hardware configuration. */
+static const struct spi_config spi_cfg =  {
+	.frequency = CONFIG_ST25R3911B_SPI_FREQ,
+	.operation = (SPI_OP_MODE_MASTER | SPI_WORD_SET(8) |
+		      SPI_TRANSFER_MSB | SPI_LINES_SINGLE |
+		      SPI_MODE_CPHA),
+	.slave = 0,
+};
+
+
+static bool reg_is_valid(u8_t reg)
+{
+	return (reg < ST25R3911B_REG_IC_IDENTITY);
+}
+
+static bool cmd_is_valid(u8_t cmd)
+{
+	return ((!((cmd >= ST25R3911B_CMD_SET_DEFAULT) && (cmd <= ST25R3911B_CMD_ANALOG_PRESET))) ||
+		(!((cmd >= ST25R3911B_CMD_MASK_RECEIVE_DATA) && (cmd <= ST25R3911B_CMD_TEST_ACCESS))));
+}
+
+static int spi_ss_ctrl(bool enable)
```

Comment:
Why don't you take advantage of the `cs` field of the `spi_config` structure? It provides the same functionality. You would only need to provide multiple buffers in the calls to `spi_transceive` instead of calling `spi_transceive` multiple times between calls to `spi_ss_ctrl`, which would be also a better approach.

Reply:
Ok